pandas/tests/io/test_pickle.py

```diff
@@ -633,3 +633,13 @@ def test_pickle_big_dataframe_compression(protocol, compression):
         partial(pd.read_pickle, compression=compression),
     )
     tm.assert_frame_equal(df, result)
+
+
+def test_pickle_frame_v124_unpickle_130():
+    # GH#42345 DataFrame created in 1.2.x, unpickle in 1.3.x
+    path = os.path.join(legacy_dirname, "1.2.4", "empty_frame_v1_2_4-GH#42345.pkl")
+    with open(path, "rb") as fd:
```

Comment:
can you also do thsi with `pd.read_pickle` (I think it works w/o your change prob)